PermanentOrg/web-app
# Show warning in app about scheduled downtime

This PR adds a component to our application which shows an announcement banner which can be used for announcing scheduled downtime within the app (and any other thing we need to announced throughout the entire app, I guess). This PR does *not* contain any data for announcements, and to test it you will have to add your own announcement. Follow the testing instructions below to figure out how to do that.

**How 2 Test**
1. Go into `src/app/announcement/data/events.ts` and add a new `AnnouncementEvent` to the currently empty array.
    * `start` represents a JS timestamp which triggers when the Announcement starts being active.
    * `end` represents a JS timestamp which triggers when the Announcement stops being active.
    * `message` is the HTML message which shows in the announcement box.
2.  For testing purposes, I like to use the following:
```js
{
  start: Math.random(), // Setting it to Random means the announcement will reappear after being dismissed
  end: Infinity, // This prevents the announcement from disappearing
  message: 'Lorem Ipsum'
}
```
3. Save the array and then launch the application. The AnnouncementComponent should appear on the login screens, the main app window, and also the public gallery.
4. Make sure to test with the current date, past dates, and a future date as well.
